### ISSUE TITLE
Misspelled spacegroup field in generated CIFs

### ISSUE THREAD

**Issue by sbliven** [biojava/biojava#480]
CIF files should contain `_symmetry.space_group_name_H-M`, but the hyphen is getting converted into an underscore. This may occur for other mmCIF fields containing hyphens as well:

```
$ sed -En 's/^\s*(save_)?(_[^ ]+-[^ ]+)/\2/p' mmcif_pdbx_v40.dic|sort -u
_diffrn_source.take-off_angle
_em_2d_crystal_entity.space_group_name_H-M
_pdbx_nmr_constraints.NA_alpha-angle_constraints_total_count
_pdbx_nmr_constraints.NA_beta-angle_constraints_total_count
_pdbx_nmr_constraints.NA_chi-angle_constraints_total_count
_pdbx_nmr_constraints.NA_delta-angle_constraints_total_count
_pdbx_nmr_constraints.NA_epsilon-angle_constraints_total_count
_pdbx_nmr_constraints.NA_gamma-angle_constraints_total_count
_pdbx_nmr_constraints.NA_other-angle_constraints_total_count
_pdbx_nmr_force_constants.non-bonded_inter_conf_db_potential_term
_pdbx_nmr_force_constants.non-bonded_inter_radius_of_gyration_term
_pdbx_nmr_force_constants.non-bonded_inter_radius_of_gyration_term_units
_pdbx_nmr_force_constants.non-bonded_inter_van_der_Waals_term
_pdbx_nmr_force_constants.non-bonded_inter_van_der_Waals_term_type
_pdbx_nmr_force_constants.non-bonded_inter_van_der_Waals_term_units
_pdbx_point_symmetry.H-M_notation
_space_group.name_H-M_alt
_symmetry.pdbx_full_space_group_name_H-M
_symmetry.space_group_name_H-M
```

See also https://github.com/arose/ngl/pull/111

**Comment by sbliven** [comment]
Unit tests are missing this because the SimpleMMcifParser internally converts all hyphens to underscores. We don't test that writing a CIF matches the original CIF, so we don't pick this up.

**Comment by josemduarte** [comment]
I've just discovered that the 2 pull requests related to this issue (#480 and #489) were not included into the `bugfixes-4.2` branch. Even if the pull request introduced some new things (e.g. annotation tags), essentially they are just fixing this bug. I'll try to cherry-pick them and produce a 4.2.3 release if there's nothing against it.

**Comment by andreasprlic** [comment]
go for it... creating a release is now only pressing a button in our new Jenkins setup!

**Comment by sbliven** [comment]
Oops! Good catch, Jose. This is definitely an important fix for the 4.2 line.
